### ISSUE TITLE
Member list "Select all" checkbox not working properly

### ISSUE THREAD

**Issue by Beamanator** [Expensify/App#20804]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to a workspace and invite few members
2. Click on select all check box and notice sometimes in the first try it works, but sometimes it doesn't. Try multiple times and see only the checkbox gets selected and not members.

## Expected Result:
Select all checkbox should be woking properly, meaning it should select everyone (except maybe not the current user / owner)

## Actual Result:
Select all checkbox not woking properly

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** `main`
**Reproducible in staging?:** N
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:** @priya-zha
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686829527636169

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by BeeMargarida** [comment]
Hi I'm Ana from Callstack - expert contributor group - I can work on this issue.

**Comment by BeeMargarida** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

After inviting new members for a workspace and then selecting all elements in the workspace invite page, the select all would stop working.

### What is the root cause of that problem?

We were saving the account IDs as numbers in the optimistic update, but then the backend would return them as strings. We were making the comparison with 2 different types, so when it should return checked true, it was returning false.

### What changes do you think we should make in order to solve the problem?

Change the `isChecked` check to convert accountID into string. The changes in server are slower to be deployed - mentioned [here](https://callstack-hq.slack.com/archives/C03UK30EA1Z/p1686822916737369?thread_ts=1686607721.733909&cid=C03UK30EA1Z) this would be the way to get this working now.

### What alternative solutions did you explore? (Optional)

--

**Comment by Beamanator** [comment]
I don't think we need to do the standard checklist here - the bug was most likely due to the HUGE onyx refactor we did a few weeks back, and we expected to break things there... So I think @slafortune we can just pay this one out and close 👍

**Comment by slafortune** [comment]
Cool!  And I think that there is only the reporting payout since this was worked on by Callstack - we are billed by them.

@priya-zha [offer sent](https://www.upwork.com/nx/wm/pre-hire/c/8577561/offer/25378106)

**Comment by priya-zha** [comment]
Thanks. Accepted.
